lib/src/main/java/com/basistheory/android/view/CardNumberElement.kt

```diff
@@ -21,14 +27,49 @@ class CardNumberElement : TextElement {
         attrs,
         defStyleAttr
     ) {
-      init()
+        init()
+    }
+
+    override fun afterTextChangedHandler(editable: Editable?) {
+        val cardResult = cardBrandEnricher.evaluateCard(getDigitsOnly(editable?.toString()))
+        if (cardResult.cardDetails?.cardMask != null)
+            mask = cardResult.cardDetails!!.cardMask.toList()
+
+        super.afterTextChangedHandler(editable)
     }
 
     private fun init() {
         super.keyboardType = KeyboardType.NUMBER
         super.mask = defaultMask
         super.transform = regexReplaceElementTransform(Regex("""\s"""), "")
         super.validate = ::luhnValidator
+
+        super.elementChangeEvent =
+            { value: String?, _: Boolean, isEmpty: Boolean, isValid: Boolean ->
+                val cardResult = cardBrandEnricher.evaluateCard(getDigitsOnly(value))
+                val eventDetails = cardResult.cardDetails?.brand?.let { b ->
+                    mutableListOf(
+                        EventDetails(
+                            "cardBrand",
+                            b
+                        )
+                    )
+                } ?: mutableListOf()
```

Comment:
I'm getting a slight code smell here around making this lambda settable, as opposed to having this be an overridable function. In detail, I think you can accomplish the same thing in more traditional way by defining it like:
```kotlin
open fun createElementChangeEvent(...): ChangeEvent {
    return ChangeEvent(
                isComplete,
                isEmpty,
                isValid,
                mutableListOf()
            )
}
```

in `TextElement` and overriding in `CardNumberElement` as:
```kotlin
open fun createElementChangeEvent(...): ChangeEvent {
    val cardResult = cardBrandEnricher.evaluateCard(getDigitsOnly(value))
                val eventDetails = cardResult.cardDetails?.brand?.let { b ->
                    mutableListOf(
                        EventDetails(
                            "cardBrand",
                            b
                        )
                    )
                } ?: mutableListOf()

    return ChangeEvent(
                    cardResult.complete ?: false,
                    isEmpty,
                    isValid,
                    eventDetails
                )
}
```